api/v1beta1/gcpmachine_types.go

```diff
@@ -66,6 +66,62 @@ const (
 	IPForwardingDisabled IPForwarding = "Disabled"
 )
 
+type SecureBootPolicy string
+
+const (
+	// SecureBootPolicyEnable enables the secure boot configuration for the GCP machine.
+	SecureBootPolicyEnable SecureBootPolicy = "Enabled"
+	// SecureBootPolicyDisable disables the secure boot configuration for the GCP machine.
+	SecureBootPolicyDisable SecureBootPolicy = "Disable"
+)
+
+type VTPMPolicy string
+
+const (
+	// VTPMPolicyEnable enables the virtualized trusted platform module configuration for the GCP machine.
+	VTPMPolicyEnable VTPMPolicy = "Enabled"
+	// VTPMPolicyDisable disables the virtualized trusted platform module configuration for the GCP machine.
+	VTPMPolicyDisable VTPMPolicy = "Disable"
+)
+
+type IntegrityMonitoringPolicy string
+
+const (
+	// IntegrityMonitoringPolicyEnable enables integrity monitoring for the GCP machine.
+	IntegrityMonitoringPolicyEnable IntegrityMonitoringPolicy = "Enabled"
+	// IntegrityMonitoringPolicyDisable disables integrity monitoring for the GCP machine.
+	IntegrityMonitoringPolicyDisable IntegrityMonitoringPolicy = "Disable"
+)
+
+// GCPShieldedInstanceConfig describes the shielded VM configuration of the instance on GCP.
+// Shielded VM configuration allow users to enable and disable Secure Boot, vTPM, and Integrity Monitoring
+type GCPShieldedInstanceConfig struct {
+	// SecureBoot Defines whether the instance should have secure boot enabled.
+	// Secure Boot verify the digital signature of all boot components, and halting the boot process if signature verification fails.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is Disabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +kubebuilder:default=Disabled
```

Comment:
Setting a default value like this isn't compatible with changing over time, normally the change over time idea is when there is no default set, we should drop one or the other

Reply:
OK